projects/plugins/jetpack/extensions/shared/components/product-management-controls/index.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * WordPress dependencies
+ */
+import { BlockControls } from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import ProductManagementInspectorControl from './inspector-control';
+import ProductManagementToolbarControl from './toolbar-control';
+import InvalidProductWarning from './invalid-product-warning';
+import StripeConnectToolbarButton from '../stripe-connect-toolbar-button';
+import { store as membershipProductsStore } from '../../../store/membership-products';
+
+import './style.scss';
+
+export default function ProductManagementControls( {
+	allowCreateOneTimeInterval = true,
+	selectedProductId = 0,
+	setSelectedProductId = () => {},
+} ) {
+	const products = useSelect(
+		select =>
+			select( membershipProductsStore ).getProducts( selectedProductId, setSelectedProductId ),
+		[]
+	);
+	const { connectUrl, isApiConnected, isSelectedProductInvalid, shouldUpgrade } = useSelect(
+		select => {
+			const { getConnectUrl, getShouldUpgrade, isApiStateConnected, isInvalidProduct } = select(
+				membershipProductsStore
+			);
+			return {
+				connectUrl: getConnectUrl(),
+				isApiConnected: isApiStateConnected(),
+				isSelectedProductInvalid: isInvalidProduct( selectedProductId ),
+				shouldUpgrade: getShouldUpgrade(),
+			};
+		}
+	);
+
+	if ( shouldUpgrade ) {
+		return null;
+	}
+
+	return (
+		<>
+			{ ! isApiConnected && !! connectUrl && (
+				<BlockControls group="block">
```

Comment:
Maybe we could add the `__experimentalShareWithChildBlocks` property here so that we can show the connect button when clicking on children of the block that uses this component (if they are using the supports __experimentalExposeControlsToChildren), WDYT?

Reply:
In line with my previous comment, I've seen that we are adding the ability to manage subscriptions but not to select subscriptions. Maybe we could also generalize (or merge with the manage component) this by using the controls we have in the Payments block:

<img width="1728" alt="Captura de pantalla 2022-03-18 a las 12 39 59" src="https://user-images.githubusercontent.com/1989914/158996972-ac4af4bd-d282-417b-a0cf-715518f7cf5c.png">

WDYT?

Reply:
> add the __experimentalShareWithChildBlocks property

@rcrdortiz I've tried it but it wasn't working immediately, and I've ignored it for now.
We should definitely try to share the controls with children blocks, I agree, but given that's not a regression, we can work on it separately.
Feel free to add a follow-up issue for it, earmarked for the next iteration. 🙂 

> adding the ability to manage subscriptions but not to select subscriptions

That's because the Premium Content approach has the selection integrated with the toolbar, which is arguably the better UX.
I'm ok with having the selection in the sidebar too, but I'd say it's a low priority for now.